sor/src/main/java/com/bazaarvoice/emodb/sor/db/DeltaIterator.java

```diff
@@ -0,0 +1,197 @@
+        _oldDelta = _next;
+        UUID changeId = getChangeId(_next);
+
+        if (_list == null) {
+            _list = Lists.newArrayListWithCapacity(numBlocks);
+        }
+
+        _list.add(_next);
+
+        for (int i = 1; i < numBlocks; i++) {
+            if (_iterator.hasNext()) {
+                _next = _iterator.next();
+                if (getChangeId(_next) == changeId) {
+                    _list.add(_next);
+                    contentSize += getValue(_next).remaining();
+                } else {
+                    // fragmented delta encountered, we must skip over it
+                    skipForward();
+                    return null;
+                }
+            } else {
+                // fragmented delta and no other deltas to skip to
+                throw new DeltaStitchingException();
+            }
+        }
+
+        skipForward();
+
+        return stitchContent(contentSize);
+    }
+
+    @Override
+    protected T computeNext() {
+        if (_next == null) {
+            return endOfData();
+        }
+
+        ByteBuffer content;
+
+        if (!_reverse) {
+
+            int numBlocks = getNumBlocks(_next);
+
+            if (numBlocks == 1) {
+                T ret = convertDelta(_next);
+                skipForward();
+                return ret;
+            }
+
+            content = compute(numBlocks);
+
+        } else {
+            if (getBlock(_next) == 0) {
+                T ret = convertDelta(_next);
+                _next = _iterator.next();
+                return ret;
+            }
+
+            content = reverseCompute();
+        }
+
+        _list.clear();
+
+        if (content == null) {
+            return computeNext();
+        }
+
+
+        return convertDelta(_oldDelta, content);
+    }
+
+    // This handles the edge case in which a client has explicity specified a changeId when writing and overwrote an exisiting delta.
+    private void skipForward() {
+        _next = null;
+        while (_iterator.hasNext() && getBlock(_next = _iterator.next()) != 0) {
+            _next = null;
+        }
+    }
+
+    // converts utf-8 encoded hex to int by building it digit by digit.
+    private int getNumBlocks(R delta) {
+        ByteBuffer content = getValue(delta);
+        int numBlocks = 0;
+
+        // build numBlocks by adding together each hex digit
+        for (int i = 0; i < _prefixLength; i++) {
+            byte b = content.get();
+            numBlocks += (b <= '9' ? b - '0' : b - 'A' + 10) << (4 * (_prefixLength - i - 1));
+        }
+
+        return numBlocks;
+    }
+
+    private ByteBuffer stitchContent(int contentSize) {
+        ByteBuffer content = ByteBuffer.allocate(contentSize);
+        for (R delta : _list) {
+            content.put(getValue(delta));
+        }
+        content.position(0);
+        return content;
+    }
+
+    abstract protected T convertDelta(R delta);
+
+    abstract protected T convertDelta(R delta, ByteBuffer content);
```

Comment:
As we spoke about, I don't think this is worth it, as it would force us to create a new StitchedRow for every delta, even ones that are already a single block